pipeline/src/resources/ColorAnnotator.py

```diff
@@ -0,0 +1,59 @@
+
+import tornado.web
+import socket
+import tornado.ioloop
+import json
+
+
+class Annotator(tornado.web.RequestHandler):
+    def __init__(self):
+        self._addr = ''
+        self._port = 12345
+        self._stream = None
+
+    def _initialize(self, sock):
+        client_socket, _ = sock.accept()
+        self._stream = tornado.iostream.IOStream(client_socket)
+        # self._stream.connect((self._addr, self._port))
+
+    def _handle_request(self, sock, fd, events):
+        self._initialize(sock)
+        self._stream.read_until(b'}', callback=self._annotate)
+
+    def _annotate(self, data):
+        resp = self.annotate(json.loads(data))
+        self.respond(resp)
+
+    def respond(self, data):
+        print(data)
+        self._stream.write(bytes(json.dumps(data), 'UTF-8'))
+
+    def start(self):
+        print('Starting annotator')
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)
+        sock.bind((self._addr, self._port))
+        sock.listen(5)
+
+        io_loop = tornado.ioloop.IOLoop.current()
+        io_loop.add_handler(sock.fileno(),
+                            lambda fd, events: self._handle_request(sock, fd, events),
+                            io_loop.READ)
+        io_loop.start()
+
+
+class ColorAnnotator(Annotator):
+    def __init__(self):
+        super().__init__()
+        self.color_words = ['red', 'blue', 'yellow']
+
+    def annotate(self, data):
+        return {
+            'color': 'red',
```

Comment:
FYI: I named the feature "name" instead of color. Maybe `color_name` would be better?

Reply:
I preferred `color` and changed yours - I don't like `name`, but would be fine with `color_name`